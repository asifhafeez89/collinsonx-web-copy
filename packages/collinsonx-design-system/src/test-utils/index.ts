import userEvent from '@testing-library/user-event';

export * from '@testing-library/react';
export { userEvent };
export { default as render, Provider } from './render';
