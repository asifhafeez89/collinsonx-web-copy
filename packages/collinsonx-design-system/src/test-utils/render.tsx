import { RenderOptions, render } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
import { ReactElement, ReactNode } from 'react';

export const Provider = ({ children }: { children: ReactNode }) => (
  <MantineProvider>{children}</MantineProvider>
);

type Render = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => ReturnType<typeof render>;

const customRender: Render = (ui, options) =>
  render(ui, {
    wrapper: Provider as unknown as RenderOptions['wrapper'],
    ...options,
  });

export default customRender;
