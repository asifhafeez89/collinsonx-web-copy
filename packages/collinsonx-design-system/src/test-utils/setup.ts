import userEvent from '@testing-library/user-event';
import render from './render';

export default function (jsx: Parameters<typeof render>[0]) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}
