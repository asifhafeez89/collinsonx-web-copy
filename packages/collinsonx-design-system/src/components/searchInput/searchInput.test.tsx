import renderer from 'react-test-renderer';
import SearchInput from '.';
import { render, screen, waitFor, userEvent, Provider } from 'test-utils';

const mockFn = jest.fn();
const mockClFn = jest.fn();
describe('<SearchInput />', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders SearchInput', () => {
    const tree = renderer
      .create(
        <Provider>
          <SearchInput
            placeholder="Search for airport or lounge"
            value={'Hello'}
            onChange={mockFn}
            onClickClear={mockClFn}
          />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('can type ', async () => {
    render(
      <Provider>
        <SearchInput
          placeholder="Search for airport or lounge"
          value={'Hello'}
          onChange={mockFn}
          onClickClear={mockClFn}
        />
      </Provider>
    );

    const inputnode = screen.getByPlaceholderText(
      'Search for airport or lounge'
    );

    userEvent.type(inputnode, 'Los Angeles');

    await waitFor(() => {
      expect(mockFn).toHaveBeenCalledTimes(11);
    });
  });
});
