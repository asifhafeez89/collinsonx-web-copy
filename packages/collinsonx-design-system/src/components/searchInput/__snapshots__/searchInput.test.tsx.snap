// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> renders SearchInput 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": "@media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}",
      }
    }
    data-mantine-styles="classes"
  />,
  <div
    className="m-46b77525 mantine-InputWrapper-root mantine-TextInput-root"
    style={{}}
  >
    <div
      className="m-6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"
      data-variant="default"
      data-with-right-section={true}
      style={{}}
    >
      <input
        aria-invalid={false}
        className="m-8fb7ebe7 mantine-Input-input mantine-TextInput-input"
        data-variant="default"
        id="mantine-"
        onChange={[MockFunction]}
        placeholder="Search for airport or lounge"
        style={
          {
            "backgroundcolor": "#fff",
            "bordercolor": "#ced4da",
            "height": "50px",
          }
        }
        value="Hello"
      />
      <div
        className="m-82577fc2 mantine-Input-section mantine-TextInput-section"
        data-position="right"
        style={{}}
      >
        <button
          aria-label="Clear"
          className="mantine-focus-auto mantine-active m-8d3f4000 mantine-ActionIcon-root m-87cf2631 mantine-UnstyledButton-root"
          onClick={[MockFunction]}
          style={
            {
              "color": "#000",
            }
          }
          type="button"
        >
          <test-file-stub />
        </button>
      </div>
    </div>
  </div>,
]
`;
